enhancements/olm-deprecate-copied-csvs.md

```diff
@@ -0,0 +1,155 @@
+
+### Goals
+
+- Allieviate the resource utilization incurred by Copied CSVs
+- Highlight features that subsume the featureset of Copied CSVs
+- Enumerate the steps necessary to deprecate and remove Copied CSVs from use
+
+### Non-Goals
+
+- Provide a zero-effort migration to clients
+
+## Proposal
+
+For each non-Copied CSV, a single [Operator](https://github.com/operator-framework/enhancements/blob/master/enhancements/simplify-olm-apis.md#proposal) resource will be generated. This cluster-scoped resource will take the place of any related Copied CSVs.
+
+### User Stories
+
+#### Feature Toggle
+
+As a Cluster Admin, I want a way to disable Copied CSVs so that I can reduce resource consumption ahead of the feature's deprecation.
+
+#### View "Available" Operators
+
+As a User, I want a way to view the operators that can act in a given namespace I have access to.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Operator Cardinality
+
+Today, Operator resources are generated on a per-Subscription basis. Since CSVs -- and thus Copied CSVs -- can exist without matching Subscriptions, OLM's controllers will be extended to generate additional Operator resources for these "free CSVs" such that:
+
+- there exists at least one`Operator` resource for every CSV
+- there exists at least one `Operator` resource for every `Subscription`
+
+That is, all CSVs installed by a given `Subscription` share an `Operator` resource __and__ any CSVs that weren't installed by a `Subscription` are related to their own `Operator` resource.
+
+#### Copied RBAC 
+
+In addition to operator discoverability, Copied CSVs also serve as convienient vehicles for the garbage collection of Copied RBAC. These are resources that are copied along with the CSV to provide the operator access to a namespace. To support the loss of Copied CSVs, the RBAC copying logic in OLM will be extended to write additional owner references from copied resources to the related Operator resource. 
+
+#### The Toggle
+
+A feature toggle will be added to control OLM's use of Copied CSVs. When enabled (toggled on), OLM will generate Copied CSVs (much like it does today). When disabled (toggled off), OLM will delete all existing Copied CSVs and will not generate any more. Since Copied CSVs are an existing feature, this toggle will be enabled by default. 
+
+The toggle will be implemented as a command-line option on the olm-operator command. 
+
+On an OpenShift cluster, Copied CSVs will be disabled by setting [a CVO override](https://docs.openshift.com/container-platform/4.8/logging/config/cluster-logging-maintenance-support.html#unmanaged-operators_cluster-logging-unsupported) for the `operator-lifecycle-manager` `ClusterOperator` and updating the command option in the olm-operator `Deployment`.
+
+> **Note:** setting a CVO override will block cluster upgrades.
```

Comment:
Could this toggle take a list of InstallMode types, such that we disable copying of _just_ CSVs with those install modes?

If I really want to disable _all_ copied CSVs, I can specify `--disable-copied-csvs=AllNamespaces,SingleNamespace,MultiNamespace`

From a customer perspective, `--disable-copied-csvs=AllNamespaces` seems like a big win because AllNamespaces CSVs are by far the worst offenders, but if I disable _only_ AllNamespaces CSV copying, I don't lose as much visibility, especially if we include Kevin's suggestion of projecting the install mode of the operator into the `Operator` object. In this scenario, I still get copied CSVs for single and multi-namespace operators, and I can go look for `Operator`s with `AllNamespaces` projected into the status to round out my view of what's watching my namespace.